cartographer/mapping/internal/2d/pose_graph_2d.cc

```diff
@@ -640,6 +702,36 @@ void PoseGraph2D::RunOptimization() {
   data_.global_submap_poses_2d = submap_data;
 }
 
+bool PoseGraph2D::ShouldSkipTrajectoryWorkItem(int trajectory_id) {
+  auto it = data_.trajectories_state.find(trajectory_id);
+  if (it == data_.trajectories_state.end()) {
+    LOG(WARNING) << "trajectory_id:" << trajectory_id
+                 << " has not been added "
+                    "but modification is requested.";
+    return false;
+  }
+  if (it->second.state == TrajectoryState::FINISHED) {
+    // TODO(gaschler): Replace all FATAL to WARNING after some testing.
+    LOG(FATAL) << "trajectory_id " << trajectory_id
+               << " has finished "
+                  "but modification is requested, skipping.";
+    return true;
+  }
+  if (it->second.deletion_state != InternalTrajectoryState::NORMAL) {
+    LOG(FATAL) << "trajectory_id " << trajectory_id
```

Comment:
Why fatal?

Reply:
nevermind, overlooked your comment `// TODO(gaschler): Replace all FATAL to WARNING after some testing.`